src/System.Data.Common/src/System/Data/Common/DbCommand.cs

```diff
@@ -123,26 +123,24 @@ public virtual Task<int> ExecuteNonQueryAsync(CancellationToken cancellationToke
             {
                 return ADP.CreatedTaskWithCancellation<int>();
             }
-            else
-            {
-                CancellationTokenRegistration registration = new CancellationTokenRegistration();
-                if (cancellationToken.CanBeCanceled)
-                {
-                    registration = cancellationToken.Register(s => ((DbCommand)s).CancelIgnoreFailure(), this);
-                }
-
-                try
-                {
-                    return Task.FromResult(ExecuteNonQuery());
-                }
-                catch (Exception e)
-                {
-                    return Task.FromException<int>(e);
-                }
-                finally
-                {
-                    registration.Dispose();
-                }
+
+            var registration = new CancellationTokenRegistration();
+            if (cancellationToken.CanBeCanceled)
```

Comment:
FWIW, I suspect this check isn't necessary.  Register does the same check, so doing such a check prior to calling Register is only useful on super hot paths where it saves the cost of the call to Register itself, loading the delegate from a field, etc.  For simplicity, consider making it just:
```C#
CancellationTokenRegistration registration = cancellationToken.Register(s => ((DbCommand)s).CancelIgnoreFailure(), this);
```